sources/engine/Stride.Engine/Engine/InstancingUserArray.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using Stride.Core;
+using Stride.Core.Mathematics;
+using Stride.Engine.Processors;
+
+namespace Stride.Engine
+{
+    [DataContract("InstancingUserArray")]
+    [Display("UserArray")]
+    public class InstancingUserArray : InstancingBase
+    {
+        /// <summary>
+        /// The instance transformation matrices.
+        /// </summary>
+        [DataMemberIgnore]
+        public Matrix[] WorldMatrices = new Matrix[0];
```

Comment:
`Array.Empty<Matrix>()` to avoid needless additional allocation here and in `WorldInverseMatrices`.

Reply:
Good one!